### ISSUE TITLE
Updating Homebrew version

### ISSUE THREAD

**Issue by LecrisUT** [pdfpc/pdfpc#590]
I'm back to update to update the Homebrew version to 4.5. I've noticed there is a new dependency and a lot of things added since 4.4 so there's quite some work to bring the dependencies on Homebrew as well.
- [ ] Bring webkitgtk to homebrew. I assume that it refers to this software: https://webkitgtk.org/
  - [x] `libicucore.dylib` missing on BigSur
  - [ ] What are the minimum configuration needed?
  - [ ] Provide tests
- [ ] Temp: build without webkitgtk features: If there's a flag to do so please let me know
- [ ] Build pdfpc 4.5
- [ ] Test the new features

**Comment by fnevgeny** [comment]
Please see #555.

> build without webkitgtk features: If there's a flag to do so please let me know

No, there is not. Webkit is currently used for rendering any kind of notes (including plain text). I'd rather expect the problem(s) with compilation of webkit-gtk under macOS to be solved soon.

**Comment by LecrisUT** [comment]
Ok, so I will go ahead and implement `webkit-gtk` as well.

**Comment by LecrisUT** [comment]
Well, `webkitgtk` needs more patching up to be able to build. There are trivial issues like needing to include other packages like `icu4c`, but now one issue is harder to pin-down: their `CMakeFiles.txt` does not pick up the appropriate headers.
```
../Source/JavaScriptCore/jsc.cpp:123:10: fatal error: 'wtf/spi/darwin/ProcessMemoryFootprint.h' file not found
#include <wtf/spi/darwin/ProcessMemoryFootprint.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
This is for the stable branch, the newest one seems to resolve this particular problem, but introduce other ones that are more tricky to resolve.
```
../Source/WebKit/Platform/SharedMemory.h:79:38: error: use of non-static data member 'm_size' of 'SharedMemory' from nested type 'Handle'
        size_t size() const { return m_size; }
                                     ^~~~~~
1 error generated.
```

So it seems that it will take some patching to get webkit-gtk to work on macOS

**Comment by storopoli** [comment]
How are this going? Would be nice to have `pdfpc` in homebrew and also specially `webkit-gtk`. @LecrisUT do you need help?

**Comment by LecrisUT** [comment]
Unfortunately I couldn't manage enough time to further debug this issue, at least for another 1.5 months. Here is the current draft I was working on.
```ruby
class Webkitgtk < Formula
  desc "Webkit port for gtk"
  homepage "https://webkitgtk.org/"
#  url "https://webkitgtk.org/releases/webkitgtk-2.31.91.tar.xz"
#  sha256 "29a60b28dfbff1e25e53f63549cf6b68b7af97c7336947e705a32f234cad834c"
  url "https://webkitgtk.org/releases/webkitgtk-2.30.6.tar.xz"
  sha256 "50736ec7a91770b5939d715196e5fe7209b93efcdeef425b24dc51fb8e9d7c1e"
  license "LGPLv2+"

  depends_on "gtk+3"
  depends_on "libsecret"
  depends_on "enchant"
  depends_on "libnotify"
  depends_on "woff2"
  depends_on "icu4c"
  depends_on "gettext"
  depends_on "cmake" => :build
  depends_on "ninja" => :build

  patch :DATA
  
  def install
    opts = %W[
         -GNinja
         -DCMAKE_C_FLAGS_RELEASE="-O3 -DNDEBUG"
         -DCMAKE_CXX_FLAGS_RELEASE="-O3 -DNDEBUG"
         -DPORT=GTK
         -DENABLE_GRAPHICS_CONTEXT_GL=OFF
         -DUSE_LIBHYPHEN=OFF
         -DUSE_SYSTEMD=OFF
         -DENABLE_GAMEPAD=OFF
         -DICU_ROOT=#{Formula["icu4c"].opt_prefix}
    ]

    mkdir "build" do
      system "cmake", "..", *std_cmake_args, *opts
      system "cmake", "--build", "."
      system "cmake", "--build", ".", "--", "install"
    end
        # system "cmake", "-S", ".", "-B", "build", *std_cmake_args
  end

  test do
    system "false"
  end
end

__END__
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0478d0c..756a06a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,10 +8,12 @@
 # and loading the cross-compilation settings from CMAKE_TOOLCHAIN_FILE.
 #
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.20)
 
 project(WebKit)
 
+find_package(ICU COMPONENTS data uc i18n REQUIRED)
+
 if (NOT CMAKE_BUILD_TYPE)
     message(WARNING "No CMAKE_BUILD_TYPE value specified, defaulting to RelWithDebInfo.")
     set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Choose the type of build." FORCE)
```

Notice the patches needed at the end of the file. There are various more still needed some in the base code itself, others in the `Cmakefile.txt`. I couldn't figure out how the build of specific components is triggered and I did not get any reply from `webkit` core team. If anyone can help escalate this issue and better phrase it, please help out. If you need a repository for it I'll make one so we can discuss more there.

**Comment by storopoli** [comment]
I will try to find someone who is more experienced than I am with CMake and build tools to help out.

**Comment by incidentist** [comment]
Hi, I found this issue while trying to make a homebrew formula for [Nyxt](https://github.com/atlas-engineer/nyxt) which also uses webkitgtk. @LecrisUT your draft formula was very helpful. There is now a webkitgtk formula in my tap at https://github.com/incidentist/homebrew-nyxt that builds and runs successfully in Big Sur. It might be enough to unblock the rest of this ticket.

**Comment by LecrisUT** [comment]
@incidentist thank you very much just a few notes. We should use `cmake test` maybe add ones if none are included. considering the other available formula names: `ios-webkit-debug-proxy`,`webkit2png` would renaming it to `webkit-gtk` be ok? Also have you checked against latest unstable if these still work?

Let me know how actual tests on your side are going, and if anyone can test on this side as well. I am still a bit too locked atm. After that could a draft PR to homebrew? Normally they would object to having these many patches, but I believe it is ok for a first submission if we also contact upstream letting them know this version is being maintained and ask them to include these patches.

**Comment by LecrisUT** [comment]
Some hints on how to do the automatic test via `cmake`. It looks like you only need to add another [`cmake --DENALBE_API_TESTS .`](https://github.com/WebKit/WebKit/blob/672b602d9df0ca12baef084a63c7810c09803de6/CMakeLists.txt#L25-L27) and most likely you'll be able to do [`ctest`](https://cmake.org/cmake/help/latest/manual/ctest.1.html). Can't find a good formula for reference but hopefully you can find how to interface with it.

Not sure if homebrew can distinguish tests for PR and build-from-source compared to bottle, but if it is we should have different levels of `ctest` for each.

**Comment by LecrisUT** [comment]
Sorry, I haven't been able to work on this. Both because I have replaced the mac machine and time constraints
